eva-accession-clustering-automation/cluster_one_assembly.py

```diff
@@ -0,0 +1,130 @@
+
+    command = 'bsub -J ' + job_name + ' -o ' + log_file + ' -e ' + error_file + ' -M 8192 -R "rusage[mem=8192]" ' + \
+              'java -jar ' + clustering_artifact + ' -Dspring.config.location=' + properties_path
+
+    print(command)
+    add_to_command_file(properties_path, command, automation_timestamp)
+    return command
+
+
+def add_to_command_file(properties_path, command, automation_timestamp):
+    """
+    This method writes the commands to a text file in the output folder. If it was run using the
+    cluster_multiple_assemblies method the automation timestamp will be provided and used. That way the commands will
+    be on the same file when ran from the automation script.
+    """
+    commands_path = os.path.dirname(properties_path) + '/commands_'
+    if automation_timestamp:
+        commands_path += automation_timestamp
+    else:
+        commands_path +=  timestamp
+    commands_path += '.txt'
+
+    with open(commands_path, 'a+') as commands:
+        commands.write(command + '\n')
+
+
+def run_clustering(source, vcf_file, project_accession, assembly_accession, private_config_file,
+                   private_config_xml_file, profile, output_directory, clustering_artifact, only_printing,
+                   automation_timestamp):
+    preliminary_check(source, vcf_file, project_accession)
+    clustering_artifact_path = get_clustering_artifact(clustering_artifact, private_config_file)
+    properties_path = create_properties_file(source, vcf_file, project_accession, assembly_accession,
+                                             private_config_file, private_config_xml_file, profile, output_directory)
+    command = generate_bsub_command(assembly_accession, properties_path, clustering_artifact_path, automation_timestamp)
+    if not only_printing:
+        run_command_with_output('Run clustering command', command, return_process_output=True)
+
+
+def preliminary_check(source, vcf_file, project_accession):
+    """
+    This checks must pass in order to run the script
+    """
+    check_valid_sources(source)
+    check_vcf_source_requirements(source, vcf_file, project_accession)
+
+
+def get_clustering_artifact(clustering_artifact_arg, private_config_file):
+    """
+    This method checks the artifact was passed to the script either using the parameter (-ca, --clustering-artifact) or
+    in the private configuration file, and gets its value. The parameter is prioritized.
+    """
+    if clustering_artifact_arg:
+        return clustering_artifact_arg
+
+    private_config_args = get_args_from_private_config_file(private_config_file)
+    if private_config_args.get('clustering_artifact'):
+        return private_config_args['clustering_artifact']
+
+    logger.error('If the clustering artifact must be provided either by the parameters (-ca, --clustering-artifact) or'
+                 'in the private configuration file')
+    sys.exit(1)
+
+
+if __name__ == "__main__":
```

Comment:
I'm not convince this brings anything over running the clustering with a list of one

Reply:
Yes, you are right. I merged the functionality so we only have one script.